satellite/notifications/endpoint.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package notifications
+
+import (
+	"context"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/rpc"
+	"storj.io/storj/pkg/rpc/rpcstatus"
+	"storj.io/storj/satellite/overlay"
+)
+
+type Endpoint struct {
+	log     *zap.Logger
+	dialer  rpc.Dialer
+	overlay *overlay.Service
+}
+
+func NewEndpoint(log *zap.Logger, dialer rpc.Dialer, overlay *overlay.Service) *Endpoint {
+	return &Endpoint{
+		log:     log,
+		dialer:  dialer,
+		overlay: overlay,
+	}
+}
+
+func (endpoint *Endpoint) Notify(ctx context.Context, req *pb.NotifyRequest) (_ *pb.NotifyResponse, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	overlayErr := func(err error) (*pb.NotifyResponse, error) {
```

Comment:
We have a errs2.Sanitizer for this.

Reply:
it might not cover overlay error empty node as it is not a class